docs/src/templates/pages/documentation/js/codestyle.hbs

```diff
@@ -0,0 +1,134 @@
+---
+title: JS Codestyle
+section-title: JS Codestyle
+docs: true
+
+navgroup: documentation
+navsub: docs-js
+navactive: js-codestyle
+---
+
+<a name="js-linting"></a><h2 class="sg-sectionHeading">Linting Rules</h2>
+
+Most of the code conventions we suggest here are enforced by ESLint, which is run as part of the gulp build tasks.
+
+We extend the [eslint-config-airbnb-base ruleset](https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb-base/rules) with our own set of JS linting rules in the [eslint-config-fozzie module](https://github.com/justeat/eslint-config-fozzie).
+
+We set this ESLint task to autofix the JS when it runs, so many of the issues found will be autofixed automatically.  If you'd like to extend this to work with your code editor of choice, most allow you to autofix files you're developing on when saved.
+
+
+<h3 class="sg-sectionHeading">Selecting DOM elements</h3>
+
+We use the [`f-dom` module](https://github.com/justeat/f-dom) to select DOM elements. This uses the ‘qwery’ selector engine behind the scenes and takes any valid CSS selector as input. It also comes with some helper functions to make DOM selection easier (see the [documentation for more details](https://github.com/justeat/f-dom#usage)).
+
+You should always use data attributes to select DOM elements using JavaScript; *Never* use classes as they should be used purely for styling. Classes are very brittle when used in JS (they break easily) – they can quite often be moved onto another element or their names changed as components styling adapts over time.  Keeping this separation helps with code refactoring as the developer then knows that changing a classname(s) won't break any behaviour set in the JS.
```

Comment:
Maybe expand upon this section with some example of how we expect the data attributes to look? [See our discussion in the UI comms repo](https://github.je-labs.com/UI/ui-comms/issues/29).

Reply:
Have added a section on this now